assets/sass/components/adminbar/_googlesitekit-wp-adminbar.scss

```diff
@@ -68,8 +68,8 @@
 
 	// Active menu styling.
 	.ab-top-menu > .googlesitekit-wp-adminbar.hover > .ab-item,
-	&:not(.mobile) .ab-top-menu > .googlesitekit-wp-adminbar:hover > .ab-item,
-	&:not(.mobile) .ab-top-menu > .googlesitekit-wp-adminbar > .ab-item:focus,
```

Comment:
These weren't showing the white background on hover for the menu item with JS disabled due to lack of specificity to override core.